packages/auth/src/authClient.ts

```diff
@@ -380,34 +383,77 @@ export class AuthClient {
 
   /**
    * Initiates passkey (WebAuthn) authentication flow.
+   *
    * This method uses WebAuthn standards to authenticate users with biometric or hardware security keys.
+   * Can be used for both new passkey creation (when credentialId is undefined) and
+   * authentication with existing passkeys (when credentialId is provided).
    *
+   * @param {string} [credentialId] - Optional credential ID for authenticating with existing passkey
    * @returns {Promise<AuthSession>} Promise that resolves to an auth session instance
-   * @throws {Error} Currently throws "Not implemented" as this method is under development
    *
    * @example
    * ```ts
+   * // New passkey authentication
    * const authSession = await authClient.loginWithPasskey();
+   *
+   * // Authenticate with existing passkey
+   * const authSession = await authClient.loginWithPasskey("existing-credential-id");
    * ```
    */
-  public async loginWithPasskey(): Promise<AuthSession> {
+  public async loginWithPasskey(credentialId?: string): Promise<AuthSession> {
     // TODO: figure out what the current passkey code is doing.
+    // For new passkey authentication, credentialId would be undefined initially
     const stamper = await this.createWebAuthnStamper({
-      credentialId: undefined,
+      credentialId: credentialId ?? undefined,
     });
     return notImplemented(stamper);
   }
 
+  /**
+   * Creates an instance of AuthSession from a previously saved authentication session state that has not expired.
+   *
+   * @param {AuthSessionState} state - The saved authentication session state
+   * @returns {Promise<AuthSession | undefined>} A promise that resolves to an AuthSession instance if the session is valid, undefined otherwise
+   */
+  public async loadAuthSessionState(
+    state: AuthSessionState,
+  ): Promise<AuthSession | undefined> {
+    const { type, expirationDateMs, user } = state;
+    if (expirationDateMs > Date.now()) {
+      if (type === "passkey") {
+        const { credentialId } = state;
+        if (!credentialId) {
+          throw new Error(
+            "Credential ID is required for passkey authentication",
+          );
+        }
+        return await this.loginWithPasskey(credentialId);
+      } else {
+        const { bundle } = state;
+        const { orgId, idToken } = user;
+        return await this.completeAuthWithBundle({
+          bundle,
+          orgId,
+          idToken,
+          authType: type,
+        });
+      }
+    }
+    return undefined;
+  }
+
   // TODO: ... and many more.
 
   private async completeAuthWithBundle({
     bundle,
     orgId,
     idToken,
+    authType,
   }: {
     bundle: string;
     orgId: string;
     idToken?: string;
+    authType: Omit<AuthType, "passkey">;
```

Comment:
i think what you want here is `Exclude`
```suggestion
    authType: Exclude<AuthType, "passkey">;
```